### ISSUE TITLE
Possible redundancy in the code

### ISSUE THREAD

**Issue by Wuziyi616** [ubc-vision/COTR#33]
Hi, I notice that when constructing the Transformer, you always return the intermediate features at [this line](https://github.com/ubc-vision/COTR/blob/bcc9ce7451443e7c8c329a51e539f6833c15695d/COTR/models/transformer.py#L216). However, after feeding them to MLP for corr regression, you only take the prediction over the last layer at [this line](https://github.com/ubc-vision/COTR/blob/bcc9ce7451443e7c8c329a51e539f6833c15695d/COTR/models/cotr_model.py#L39). So I guess maybe you can set `return_intermediate=False` to save some memory/computation?

**Comment by Wuziyi616** [comment]
I have another (maybe stupid) question. In your positional embedding [here](https://github.com/ubc-vision/COTR/blob/bcc9ce7451443e7c8c329a51e539f6833c15695d/COTR/models/position_encoding.py#L66-L67), the order of x/y seems to be reversed? If I concat them by `xy_embed = torch.stack([x_embed, y_embed],dim=-1)`, and print their values:

```
>>> xy_embed.shape
torch.Size([2, 8, 8, 2])  # this is fine
>>> xy_embed[0, 0, 0]
tensor([1., 1.])  # this is also fine
>>> xy_embed[0, 0, 1]
tensor([2., 1.])  # shouldn't it be [1., 2.]?
>>> xy_embed[0, 5, 1]
tensor([2., 6.])  # shouldn't it be [6., 2.]?
```
I find their values a bit confusing. Any insight here?

**Comment by jiangwei221** [comment]
Hey, yes, it sounds a good idea to set `return_intermediate=False`. I would appreciate it if you can create a PR.
I'm not sure about the positional encoding question, will the permute function fix this ordering problem?
